apps/crn-server/test/integration/contentful/users.integration-test.ts

```diff
@@ -0,0 +1,172 @@
+import Chance from 'chance';
+import supertest from 'supertest';
+import { Express } from 'express';
+import { v4 as uuid } from 'uuid';
+
+import { addLocaleToFields, getLinkEntity } from '@asap-hub/contentful';
+import { UserResponse } from '@asap-hub/model';
+
+import { appFactory } from '../../../src/app';
+import { getEnvironment, TestEnvironment } from '../../helpers/contentful';
+import { retryable } from '../../helpers/retryable';
+
+jest.mock('../../../src/config', () => ({
+  ...jest.requireActual('../../../src/config'),
+  logLevel: 'silent',
```

Comment:
Because the only logs that are a) messive, and b) for the 403/404s which are _expected_ by the tests. It's not useful output and only serves to make the actual test output less readable.

```
2023-06-21T13:16:03.270Z	ERROR	Not Found	{"level":"error","time":1687353363269,"pid":42558,"hostname":"Leonards-MacBook-Pro-3.local","req":{"id":3,"method":"GET","url":"/users/3NH0bvewYMP1ur63VsBHkv","query":{},"params":{},"headers":{"host":"127.0.0.1:55905","accept-encoding":"gzip, deflate","connection":"close"},"remoteAddress":"::ffff:127.0.0.1","remotePort":55906},"err":{"type":"Error","message":"Not Found","stack":"Error: Not Found\n    at /Users/lenny/dev/asap-hub/apps/crn-server/src/routes/user.route.ts:62:27\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)","data":null,"isBoom":true,"isServer":false,"output":{"statusCode":404,"payload":{"statusCode":404,"error":"Not Found","message":"Not Found"},"headers":{}}},"msg":"Not Found"}
2023-06-21T13:16:11.611Z	ERROR	Forbidden	{"level":"error","time":1687353371606,"pid":42558,"hostname":"Leonards-MacBook-Pro-3.local","req":{"id":8,"method":"PATCH","url":"/users/mqULnUmMXdN4fpla251jg","query":{},"params":{},"headers":{"host":"127.0.0.1:55938","accept-encoding":"gzip, deflate","content-type":"application/json","content-length":"23","connection":"close"},"remoteAddress":"::ffff:127.0.0.1","remotePort":55939},"err":{"type":"Error","message":"Forbidden","stack":"Error: Forbidden\n    at /Users/lenny/dev/asap-hub/apps/crn-server/src/routes/user.route.ts:111:27\n    at newFn (/Users/lenny/dev/asap-hub/.yarn/__virtual__/express-async-errors-virtual-c5b601ee87/0/cache/express-async-errors-npm-3.1.1-c0a6b1e552-3a2a6df8ea.zip/node_modules/express-async-errors/index.js:16:20)\n    at Layer.handle [as handle_request] (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/layer.js:95:5)\n    at next (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/route.js:144:13)\n    at Route.dispatch (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/route.js:114:3)\n    at newFn (/Users/lenny/dev/asap-hub/.yarn/__virtual__/express-async-errors-virtual-c5b601ee87/0/cache/express-async-errors-npm-3.1.1-c0a6b1e552-3a2a6df8ea.zip/node_modules/express-async-errors/index.js:16:20)\n    at Layer.handle [as handle_request] (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/layer.js:95:5)\n    at /Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:284:15\n    at param (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:365:14)\n    at param (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:376:14)\n    at Function.process_params (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:421:3)\n    at next (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:280:10)\n    at Function.handle (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:175:3)\n    at router (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:47:12)\n    at newFn (/Users/lenny/dev/asap-hub/.yarn/__virtual__/express-async-errors-virtual-c5b601ee87/0/cache/express-async-errors-npm-3.1.1-c0a6b1e552-3a2a6df8ea.zip/node_modules/express-async-errors/index.js:16:20)\n    at Layer.handle [as handle_request] (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/layer.js:95:5)\n    at trim_prefix (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:328:13)\n    at /Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:286:9\n    at Function.process_params (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:346:12)\n    at next (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:280:10)\n    at authHandler (/Users/lenny/dev/asap-hub/apps/crn-server/test/integration/contentful/users.integration-test.ts:44:9)\n    at newFn (/Users/lenny/dev/asap-hub/.yarn/__virtual__/express-async-errors-virtual-c5b601ee87/0/cache/express-async-errors-npm-3.1.1-c0a6b1e552-3a2a6df8ea.zip/node_modules/express-async-errors/index.js:16:20)\n    at Layer.handle [as handle_request] (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/layer.js:95:5)\n    at trim_prefix (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:328:13)\n    at /Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:286:9\n    at Function.process_params (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:346:12)\n    at next (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:280:10)\n    at /Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:646:15\n    at next (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:265:14)\n    at Function.handle (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:175:3)\n    at router (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:47:12)\n    at newFn (/Users/lenny/dev/asap-hub/.yarn/__virtual__/express-async-errors-virtual-c5b601ee87/0/cache/express-async-errors-npm-3.1.1-c0a6b1e552-3a2a6df8ea.zip/node_modules/express-async-errors/index.js:16:20)\n    at Layer.handle [as handle_request] (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/layer.js:95:5)\n    at trim_prefix (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:328:13)\n    at /Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:286:9\n    at Function.process_params (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:346:12)\n    at next (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:280:10)\n    at /Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:646:15\n    at next (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:265:14)\n    at Function.handle (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:175:3)\n    at router (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:47:12)\n    at newFn (/Users/lenny/dev/asap-hub/.yarn/__virtual__/express-async-errors-virtual-c5b601ee87/0/cache/express-async-errors-npm-3.1.1-c0a6b1e552-3a2a6df8ea.zip/node_modules/express-async-errors/index.js:16:20)\n    at Layer.handle [as handle_request] (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/layer.js:95:5)\n    at trim_prefix (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:328:13)\n    at /Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:286:9\n    at Function.process_params (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:346:12)\n    at next (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:280:10)\n    at /Users/lenny/dev/asap-hub/apps/crn-server/src/middleware/feature-flag.ts:14:10\n    at newFn (/Users/lenny/dev/asap-hub/.yarn/__virtual__/express-async-errors-virtual-c5b601ee87/0/cache/express-async-errors-npm-3.1.1-c0a6b1e552-3a2a6df8ea.zip/node_modules/express-async-errors/index.js:16:20)\n    at Layer.handle [as handle_request] (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/layer.js:95:5)\n    at trim_prefix (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:328:13)\n    at /Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:286:9\n    at Function.process_params (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:346:12)\n    at next (/Users/lenny/dev/asap-hub/.yarn/cache/express-npm-4.18.2-bb15ff679a-3c4b9b0768.zip/node_modules/express/lib/router/index.js:280:10)\n    at /Users/lenny/dev/asap-hub/.yarn/cache/body-parser-npm-1.20.1-759fd14db9-f1050dbac3.zip/node_modules/body-parser/lib/read.js:137:5\n    at AsyncResource.runInAsyncScope (node:async_hooks:203:9)\n    at invokeCallback (/Users/lenny/dev/asap-hub/.yarn/cache/raw-body-npm-2.5.1-9dd1d9fff9-5362adff15.zip/node_modules/raw-body/index.js:231:16)\n    at done (/Users/lenny/dev/asap-hub/.yarn/cache/raw-body-npm-2.5.1-9dd1d9fff9-5362adff15.zip/node_modules/raw-body/index.js:220:7)\n    at IncomingMessage.onEnd (/Users/lenny/dev/asap-hub/.yarn/cache/raw-body-npm-2.5.1-9dd1d9fff9-5362adff15.zip/node_modules/raw-body/index.js:280:7)\n    at IncomingMessage.emit (node:events:525:35)\n    at IncomingMessage.emit (node:domain:489:12)\n    at endReadableNT (node:internal/streams/readable:1358:12)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","data":null,"isBoom":true,"isServer":false,"output":{"statusCode":403,"payload":{"statusCode":403,"error":"Forbidden","message":"Forbidden"},"headers":{}}},"msg":"Forbidden"}
 PASS  apps/crn-server/test/integration/contentful/users.integration-test.ts (27.911 s)
  users
    ✓ can fetch a list of users (415 ms)
    ✓ can fetch user (1178 ms)
    ✓ cannot fetch user if they are not onboarded (1936 ms)
    ✓ can fetch a user by connection code (910 ms)
    ✓ can fetch the groups for a user (3123 ms)
    ✓ can patch the logged-in user (2613 ms)
    ✓ cannot patch a non-logged-in user (1694 ms)

Test Suites: 1 passed, 1 total
Tests:       7 passed, 7 total
Snapshots:   0 total
Time:        27.982 s
Ran all test suites.
```